v3/how-to-guides/08-tools/d-subwasm/index.mdx

```diff
@@ -0,0 +1,104 @@
+This guide steps through the different capabilities that Subwasm provides. 
+
+## Use cases
+
+- Expose verifiable historical information of a chain's runtime for front end applications.
+- Get a quick snapshot of the modules in a runtime.
+- Check for changes between runtimes using a diff.
+- Expose a verifiable runtime for governance transparency.
+
+## Before you begin
+
+To follow this guide you should have a runtime from a chain you wish to query. 
+This could either be one running locally or one running in production.
+In order to filter metadata, you will also need to [install JQ](https://stedolan.github.io/jq/download/).
+
+## Install and discover the CLI tool
+
+To get started, install Subwasm and explore its commands.
+
+```bash
+cargo install --locked --git https://github.com/chevdor/subwasm --tag v0.16.1
+```
+Make sure you're using the [latest release](https://github.com/chevdor/subwasm/releases). 
+This guide is based on version 0.14.1. 
+
+Explore the CLI commands.
+
+```bash
+subwasm --help
+```
+
+## Get information about a runtime 
+
+1. Launch a local node from inside your chain's directory. 
+   This example is launching the node template in dev mode.
+   
+    ```bash
+    ./target/release/node-template --tmp --dev
+    ```
+
+1. In a new terminal, create a temporary folder and use Subwasm to get and save the Wasm of your local chain.
+
+    ```bash
+    mkdir TempWasms && cd TempWasms
+    subwasm get ws://localhost:9944 -o mychain.wasm
+    ```
+
+1. Run the following command to save your chain's metadata to a JSON file:
+
+   ```bash
+    subwasm --json meta mychain.wasm | jq 'del( .. | .documentation? )' > mychain-metadata.json
+   ```
+
+   Using JQ is a useful tool to filter the metadata details from the `subwasm --json meta mychain.wasm` command.
+   In this case, we're choosing to ignore all documentation metadata.
+   You can have a look at the different metadata fields a Substrate chain exposes [here](https://polkadot.js.org/docs/substrate).
+
+1. Use the custom metadata file in a front-end application. 
+
+## Compare two different runtimes
+
+Using Subwasm, we can also compare the runtimes of a chain at different blocks. 
+In this example, we'll take Polkadot's runtime at block 500 and compare it with its latest runtime.
+
+1. Go to a block explorer and retrieve the block hash for the runtime your want to compared against.
+
+    For example, go to [PolkadotJS Apps](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/explorer) and retrieve the block hash at [block 50000](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/explorer/query/50000).
+
+    `0xfdeab19649426aadbdab046c9b5dac0a5b0c850b8fd0927e3b4e6c896d5fb0b7`
+
+1. Save the Wasm blob of a previous state.
+
+    ```bash
+    subwasm get --chain polkadot --block 0xfdeab19649426aadbdab046c9b5dac0a5b0c850b8fd0927e3b4e6c896d5fb0b7 -o polkadot-500000.wasm
```

Comment:
how does subwasm get this info? What node endpoint? Can I specify that?

Reply:
It has a `get_chain_urls` function that takes a chain URL and returns the node endpoint. It'll pick whichever connects first. See [here](https://github.com/chevdor/subwasm/blob/master/lib/src/chain_urls.rs#L6-L9).

Reply:
cool! can we add this language to the page?